### DISCUSSION TITLE
Flowdeck not being detected for some reason

### DISCUSSION THREAD

**Discussion by akshaysunkara** [bitcraze/discussions#1473] [Q&A]
![IMG_4926](https://github.com/user-attachments/assets/d92c8a75-e72c-4351-b332-8eaf21c38c70)
![IMG_4927](https://github.com/user-attachments/assets/6e43cdae-04ad-4162-8032-acd8b969c7f5)
![IMG_4928](https://github.com/user-attachments/assets/f678aa55-de2f-4267-bea2-94c0a66380f0)
![IMG_4925](https://github.com/user-attachments/assets/149a0efa-d7bc-4573-b0e7-32e03bf919e6)

These are images of how the flowdeck is attached.

The flowdeck's black port thing is a little tilted, don't know if thats a problem

For some reason, its not being detected by the cfclient software

SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x313232313232471200280023 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly

However, whenever I run bootloader to flash the latest firmware, it says this towards the end:

Identifying deck to be updated! Deck updated!

Just wondering what the problem is. 

Also, I have another flowdeck that is damaged. Any ways I can fix it through soldering?

![image](https://github.com/user-attachments/assets/753e249e-0160-4be6-89bb-ec28c24c1d4d)

**Comment by evoggy** [comment]
The first issue with the Flow deck not being detected sounds a lot like an expansion connector issue, the pin for the one-wire is not connected and possible also more pins. Unfortunately the fact that it says that it's finished updating the decks during flashing doesn't mean it detects the deck.

For the second issue, yes this is possible. There's spare connects which you can buy [here](https://store.bitcraze.io/collections/spare-parts-crazyflie-2-0/products/female-deck-connector) and replace it with.

**Reply by akshaykumarsunkara** [reply]
For the first issue, is it the expansion connector of the flow deck? Could I see pictures of what is not connected to get a better idea of what to fix.

how would I go about fixing this issue?

for the second issue, do I need to buy a replacement when I already have the connector, just separated? Can I just use the same one

**Comment by NageshEranki** [comment]
Hi

In your first image, it looks like you have the flow deck attached on backwards? You will find arrows on the Crazyflie and the Flow deck showing you which way the flow deck should be aligned when mounting it.


[instructions on mounting expansion decks](https://www.bitcraze.io/documentation/tutorials/getting-started-with-expansion-decks/)

**Reply by akshaysunkara** [reply]
No, I'm pretty sure I have mounted the correct way. 

I'll double check again though

**Reply by knmcguire** [reply]
Hi! @NageshEranki is absolutely right, it needs to be the other way.

Please check this tutorial: https://www.bitcraze.io/documentation/tutorials/getting-started-with-expansion-decks/

**Reply by akshaysunkara** [reply]
Thanks a lot! Just fixed it.
